src/algebra/monoid_algebra.lean

```diff
@@ -703,55 +736,73 @@ end algebra
 
 section lift
 
-/-- Any monoid homomorphism `multiplicative G →* A` can be lifted to an algebra homomorphism
-`add_monoid_algebra k G →ₐ[k] A`. -/
-def lift [comm_semiring k] [add_monoid G] {A : Type u₃} [semiring A] [algebra k A] :
-  (multiplicative G →* A) ≃ (add_monoid_algebra k G →ₐ[k] A) :=
-{ inv_fun := λ f, ((f : add_monoid_algebra k G →+* A) : add_monoid_algebra k G →* A).comp (of k G),
-  to_fun := λ F, {
-    -- The proofs here are almost identical to `monoid_algebra.lift`, but use `erw` instead of `rw`
-    -- to unfold `multiplicative`
-    to_fun := λ f, f.sum (λ a b, b • F a),
-    map_one' := by { rw [one_def, sum_single_index, one_smul], erw [F.map_one], apply zero_smul },
-    map_mul' := λ f g,
-      begin
-        rw [mul_def, finsupp.sum_mul, finsupp.sum_sum_index],
-        work_on_goal 1 { intros, rw zero_smul, },
-        work_on_goal 1 { intros, rw add_smul, },
-        refine finset.sum_congr rfl (λ a ha, _),
-        simp only,
-        rw [finsupp.mul_sum, finsupp.sum_sum_index],
-        work_on_goal 1 { intros, rw zero_smul, },
-        work_on_goal 1 { intros, rw add_smul, },
-        refine finset.sum_congr rfl (λ a' ha', _),
-        simp only,
-        rw [sum_single_index],
-        erw [F.map_mul],
-        rw [algebra.mul_smul_comm, algebra.smul_mul_assoc, smul_smul, mul_comm],
-        apply zero_smul,
-      end,
-    map_zero' := sum_zero_index,
-    map_add' := λ f g,
-      begin
-        rw [sum_add_index],
-        { intros, rw zero_smul, },
-        { intros, rw add_smul, },
-      end,
-    commutes' := λ r,
-      begin
-        rw [coe_algebra_map, sum_single_index],
-        erw [F.map_one],
-        rw [algebra.smul_def, mul_one, algebra.id.map_eq_self],
-        apply zero_smul
-      end, },
-  left_inv := λ f, begin ext x, simp [sum_single_index] end,
+variables [comm_semiring k] [add_monoid G]
+variables {A : Type u₃} [semiring A] [algebra k A]
+variables {B : Type*} [semiring B] [algebra k B]
+
+/--
+Auxiliary definition used to implement lift
+
+For a `add_monoid_algebra A G`, this is an algebra morphism that applies `H` to the `A` part and `F` to
+the `G` part. Multiplication between `H a` and `F g` must be commutative.
+-/
+def lift_aux (F : multiplicative G →* B) (H : A →ₐ[k] B) (FH_comm : ∀ a g, H a * F g = F g * H a) : add_monoid_algebra A G →ₐ[k] B :=
+{ to_fun := λ f, f.sum (λ g a, H a * F g),
+  map_one' := by {
```

Comment:
```suggestion   
  -- The proofs here are almost identical to `monoid_algebra.lift_aux`, but use `erw` instead of `rw`	
  -- to unfold `multiplicative`
  map_one' := by {
```